components/builder-db/src/pool.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ops::{Deref, DerefMut};
+use std::result;
+use std::sync::atomic::{AtomicUsize, Ordering, ATOMIC_USIZE_INIT};
+use std::thread;
+use std::time::Duration;
+
+use r2d2;
+use r2d2_postgres::{self, PostgresConnectionManager, TlsMode};
+use postgres::transaction::Transaction;
+use postgres;
+
+use error::{Error, Result};
+
+// We will use this to allocate test schmeas
+static GLOBAL_SCHEMA_COUNT: AtomicUsize = ATOMIC_USIZE_INIT;
+
+#[derive(Debug, Clone)]
+pub struct Pool {
+    inner: r2d2::Pool<PostgresConnectionManager>,
+}
+
+impl Pool {
+    pub fn new(connection_url: &str,
```

Comment:
A constructor that sleeps the calling thread is a little awkward. I might suggest moving this off to a "start" function or something along those lines